packages/@okta/vuepress-site/use_cases/event_hooks/index.md

```diff
@@ -0,0 +1,139 @@
+
+[https://developer.okta.com/docs/api/resources/event-types/?q=event-hook-eligible](/docs/api/resources/event-types/?q=event-hook-eligible)
+
+## Requests Sent by Okta
+
+The requests sent from Okta to your external service are HTTPS requests. POST requests are used for the ongoing delivery of events, and a one-time GET request is used for verifying your endpoint.
+
+### One-Time Verification Request
+
+After registering an event hook, but before you can use it, you need to have Okta make a one-time GET verification request to your endpoint, passing your service a verification value that it needs to send back. This serves as a test confirming that you control the endpoint.
+
+This one-time verification request is the only GET request Okta will send to your external service, while the ongoing requests to notify your service of event occurrences will be HTTPS POST requests. Your web service can use the GET versus POST distinction to implement logic to handle this special one-time request.
+
+The way your service needs to handle this one-time verification is as follows: The request from Okta will contain an HTTP header named `X-Okta-Verification-Challenge`. Your service needs to read the value of that header and return it in the response body, in a JSON object named `verification`, i.e.: `{ "verification" : "value_from_header" }`. Note that the value comes to you in an HTTP header, but you need to send it back in a JSON object.
+
+### Ongoing Event Delivery
+
+After verification, for ongoing notification of events, Okta sends HTTPS POST requests to your service. The JSON payload of these requests is where Okta provides specific information about events that have occurred.
+
+Information is encapsulated in the JSON payload in the `data.events` object. The `data.events` object is an array, in order to allow sending multiple events in a single POST request. Events that occur within a short time of each other will be amalgamated, with each each element of the array providing information on one event.
+
+ The content of each array element is an object of the [LogEvent](/docs/api/resources/system_log/#example-logevent-object) type. This is the same object that the [System Log API](/docs/api/resources/system_log/) defines for System Log events. See the documentation there for information on the object and its sub-objects.
+
+ Delivery of events is best effort. Events are delivered at least once. Delivery may be delayed by network conditions. In some cases, multiple requests may arrive at the same time after a delay, or events may arrive out of order. To establish ordering, you can use the time stamp contained in the `data.events.published` property of each event.
+
```

Comment:
This section on "Requests sent by Okta" is very well written :).  

Under "Ongoing Event Delivery", we need to include details on how to troubleshoot in the event of event hook delivery failure (this can possibly be a separate section). For now, we can mention something on the lines of this:

> As mentioned, there is a very slight possibility that the event might not get delivered or may be delayed due to network conditions. We recommend checking the System Log event event_hook.delivery to identify delivery failures. To fetch missing events, we suggest using the System Log API to pull the undelivered events.

Reply:
I noticed that, in general, we do not list the Syslog events for a feature in the developer docs for that feature. But it would be nice if we start doing so. IMHO, without knowing which Syslog events to track, the customers will find it even more difficult to troubleshoot. So, at least for Event Hooks, we should list the related events or ask customers to visit the Event Types link 
https://developer.okta.com/docs/api/resources/event-types/?q=event_hook

Reply:
@ankurdatta-okta, I'm hesitating regarding adding your recommendation to check the System Log API. If this is a case of a developer debugging their event hook handler, wouldn't it be easier for them to go to the System Log UI in Admin Console? But, if you mean adding code in the their implementation, to regularly poll the System Log API, doesn't that defeat the purpose of Event Hooks?

Reply:
> @ankurdatta-okta, I'm hesitating regarding adding your recommendation to check the System Log API. If this is a case of a developer debugging their event hook handler, wouldn't it be easier for them to go to the System Log UI in Admin Console? But, if you mean the developer adding code in the implementation, to regularly poll the System Log API, doesn't that defeat the purpose of Event Hooks?

I understand and agree with your concern. I'll provide some more context which will hopefully clarify things. 

Some time later this year, the Extensibility team will start supporting a feature called "Replay" which would allow Okta to go back to the history of event hooks delivered in the past and then allow the customer to ask Okta to resend the hooks for which there were delivery failures. At present, we do not support this functionality. As a result, we have to ask customers to use the System Log API to poll for events when the events get missed or undelivered. 

--
An example of a Replay feature:
https://pinpayments.com/developers/api-reference/webhooks#put-webhook-replay